hazelcast/src/main/java/com/hazelcast/map/EntryLoader.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map;
+
+import com.hazelcast.core.MapLoader;
+
+import static com.hazelcast.util.Preconditions.isNotNull;
+
+/**
+ * This is an extension to {@link MapLoader}. Entries loaded by EntryLoader
+ * may have expiration times attached to them. In that case, the entry
+ * is removed after given expiration time.
+ *
+ * See {@link MapLoader}.
+ *
+ * @param <K> type of the EntryLoader key
+ * @param <V> type of the EnyryLoader value
+ */
+public interface EntryLoader<K, V> extends MapLoader<K, EntryLoader.MetadataAwareValue<V>> {
+
+    /**
+     * Represents a value with an expiration time attached to it.
+     * Expiration time is optional.
+     * @param <V> the type of the value object
+     */
+    class MetadataAwareValue<V> {
```

Comment:
minor: Maybe we can make `MetadataAwareValue` a standalone class since it is also used while storing.

Reply:
:)

Reply:
Do you see any technical problem with having it inside EntryLoader? I actually think it make more sense to keep it nested in MapLoader since it is strictly related to it from user's perspective.

Reply:
No technical issue i see, also, since MapStore is also a MapLoader, no big problem from this perspective. But above comment was just my expectation.